crates/synthesis/src/multi_controlled/mod.rs

```diff
@@ -10,11 +10,71 @@
 // copyright notice, and modified files need to carry a notice indicating
 // that they have been altered from the originals.
 
+use mcx::{c3x, c4x, synth_mcx_n_dirty_i15, synth_mcx_noaux_v24};
 use pyo3::prelude::*;
+use qiskit_circuit::circuit_data::CircuitData;
 
 mod mcmt;
+mod mcx;
+
+/// Synthesize a multi-controlled X gate with :math:`k` controls based on the paper
```

Comment:
It might be better not to duplicate the docs here: the Rust function is `synth_mcx_n_dirty_i15`, which is documented already, and the Python function is also documented already separately in the Python space 🙂